packages/core/cache/src/LMDBCache.js

```diff
@@ -40,45 +44,77 @@ export class LMDBCache implements Cache {
   }
 
   has(key: string): Promise<boolean> {
-    return Promise.resolve(this.store.get(key) != null);
+    if (this.store.get(key) != null) return Promise.resolve(true);
+    return this.hasLargeBlob(key);
   }
 
-  get<T>(key: string): Promise<?T> {
-    let data = this.store.get(key);
-    if (data == null) {
-      return Promise.resolve(null);
-    }
-
-    return Promise.resolve(deserialize(data));
+  async get<T>(key: string): Promise<?T> {
+    let data = await this.getBuffer(key);
+    return data == null ? null : deserialize(data);
   }
 
   async set(key: string, value: mixed): Promise<void> {
-    await this.store.put(key, serialize(value));
+    await this.setBlob(key, serialize(value));
   }
 
   getStream(key: string): Readable {
-    return blobToStream(this.store.get(key));
+    let buf = this.store.get(key);
+    if (buf != null) return blobToStream(buf);
+    return this.fs.createReadStream(path.join(this.dir, key));
   }
 
   async setStream(key: string, stream: Readable): Promise<void> {
-    let buf = await bufferStream(stream);
-    await this.store.put(key, buf);
+    await this.setBlob(key, await bufferStream(stream));
   }
 
-  getBlob(key: string): Promise<Buffer> {
-    let buffer = this.store.get(key);
-    return buffer != null
-      ? Promise.resolve(buffer)
-      : Promise.reject(new Error(`Key ${key} not found in cache`));
+  async getBlob(key: string): Promise<Buffer> {
+    let buffer = await this.getBuffer(key);
+    if (buffer == null) throw new Error(`Key ${key} not found in cache`);
+    return buffer;
   }
 
   async setBlob(key: string, contents: Buffer | string): Promise<void> {
-    await this.store.put(key, contents);
+    if (isLargeBlob(contents)) {
+      // Remove the old blob if it has been 'upgraded' to large blob storage.
+      if (this.store.get(key) != null) await this.store.remove(key);
+      await this.setLargeBlob(key, contents);
+    } else {
+      await this.store.put(key, contents);
+    }
+  }
+
+  async getBuffer(key: string): Promise<?Buffer> {
+    let buffer = this.store.get(key);
+    if (buffer == null && (await this.hasLargeBlob(key))) {
```

Comment:
I'm worried that this will result in a lot of extra `stat` calls to the filesystem since it will be checked for every key that is missing from the cache (everything on initial builds). Can we keep checking for large blobs explicitly opt-in? I thought we only needed it for certain known large entries (e.g. graphs)?

Reply:
I reverted this in favor marking large assets for streaming to and from cache ([comment](https://github.com/parcel-bundler/parcel/pull/7198#issuecomment-972389465))